tests/test_strong_branching.py

```diff
@@ -0,0 +1,201 @@
+from pyscipopt import Model, Branchrule, SCIP_RESULT, quicksum, SCIP_PARAMSETTING
+
+"""
+This is a test for strong branching. It also gives a basic outline of what a function that imitates strong
+branching should do. An example of when to use this would be to access strong branching scores etc. 
+
+If this was done fully, then one would need to handle the following cases:
+- What happens if a new primal solution is found and the bound is larger than the cutoff bound? Return CUTOFF status
+- What happens if the bound for one of the children is above a cutoff bound? Change variable bound
+- If probing is ever enabled then one would need to handle new found bounds appropriately.
+"""
+
+
+class StrongBranchingRule(Branchrule):
+
+    def __init__(self, scip, idempotent=False):
+        self.scip = scip
+        self.idempotent = idempotent
+
+    def branchexeclp(self, allowaddcons):
+
+        # Get the branching candidates. Only consider the number of priority candidates (they are sorted to be first)
+        # The implicit integer candidates in general shouldn't be branched on. Unless specified by the user
+        # npriocands and ncands are the same (npriocands are variables that have been designated as priorities)
+        branch_cands, branch_cand_sols, branch_cand_fracs, ncands, npriocands, nimplcands = self.scip.getLPBranchCands()
+
+        # Initialise scores for each variable
+        scores = [-self.scip.infinity() for _ in range(npriocands)]
+        down_bounds = [None for _ in range(npriocands)]
+        up_bounds = [None for _ in range(npriocands)]
+
+        # Initialise placeholder values
+        num_nodes = self.scip.getNNodes()
+        lpobjval = self.scip.getLPObjVal()
+        lperror = False
+        best_cand_idx = 0
+
+        # Start strong branching and iterate over the branching candidates
+        self.scip.startStrongBranching()
+        for i in range(npriocands):
+
+            # Check the case that the variable has already been strong branched on at this node.
+            # This case occurs when events happen in the node that should be handled immediately.
+            # When processing the node again (because the event did not remove it), there's no need to duplicate work.
+            if self.scip.getVarStrongBranchLastNode(branch_cands[i]) == num_nodes:
```

Comment:
From the comments, I don't understand why we are comparing to the number of nodes instead of the current node number.

Reply:
The number of nodes is the current node number (if there's been a restart or something then only `getNTotalNodes` continues to increase)

Reply:
Oh, I assumed that SCIP could revisit a prior node during the search. No problem, then